src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -19,6 +19,7 @@ internal partial class SslStreamInternal
         private static readonly AsyncProtocolCallback s_resumeAsyncReadCallback = new AsyncProtocolCallback(ResumeAsyncReadCallback);
         private static readonly AsyncProtocolCallback s_readHeaderCallback = new AsyncProtocolCallback(ReadHeaderCallback);
         private static readonly AsyncProtocolCallback s_readFrameCallback = new AsyncProtocolCallback(ReadFrameCallback);
+        private static readonly SemaphoreSlim s_throttle = new SemaphoreSlim(Environment.ProcessorCount);
```

Comment:
Have you experimented with this value? I'm wondering if some small multiple (2 or 3) would test better or worse.

Reply:
Also, can you add a comment explaining why this exists? Presumably we'll undo this once we upgrade to OpenSSL 1.1. if there's an issue tracking that, we can include its issue number here.

Reply:
Will add a comment and issue number. 

As to the multiplier, I can retry these numbers I believe that actually a smaller number (/2) was better on boxes with hyperthreading as actually the aes algo uses pretty much add the registers/pipeline anyway. As azure doesn't do hyperthreading (well some newer instances do) 1x was the happy medium. 

But that is from memory so I will try to retest on physical vs azure, I am limited to 16 cores on azure and my physical has 24 cores so it's also a bit Apple's and oranges.

Reply:
> I can retry these numbers I believe that actually a smaller number (/2) 

I could believe that as well, but that's also dangerous, especially when using the same logic on Windows.

> There isn't a way that I know of to get the "real" core count rather than virtual without me going to native apis...?

Correct; that's not currently exposed anywhere.  There have been discussions in the past about exposing more about the machine's topology, but that's never materialized in built-in APIs.  I'm fairly certain there are 3rd party implementations of that, though, including if I remember correctly in Joe Duffy's concurrency book.

Reply:
Its also more complicated with Jobs/Docker/cg_groups etc

Reply:
Yeah I have my own code for that for some stuff I do at work (heavy handed avx montecarlo simulations are a good place to ignore hyperthreads to avoid context thrashing ;)) but I didn't want to include that here. Maybe I can put in an API review. Anyway I will run a basic test at 2x and see what direction it heads

Reply:
As aside have a thing for Windows that does Logical vs Physical https://github.com/davidfowl/Channels/blob/master/src/Channels.Networking.Windows.RIO/Internal/CpuInfo.cs#L31-L43

Reply:
Me 3 and linux, but I figure that might be pushing the boundary for this PR ;)

Reply:
> it goes from ~720k to ~200k-300k

Ok. Thanks for checking.

> that might be pushing the boundary for this PR

Yes ;)